src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultGeneratedDocumentPublisher.cs

```diff
@@ -68,7 +68,7 @@ public override void PublishCSharp(string filePath, SourceText sourceText, long
             IReadOnlyList<TextChange> textChanges = Array.Empty<TextChange>();
             if (!sourceText.ContentEquals(previouslyPublishedData.SourceText))
```

Comment:
@NTaylorMullen, do we still need this check here? We check the same thing again in that method.

Reply:
Hmm we can get rid of it but we'll need to adjust the else if portion to understand that 0 text changes && hostDocumentVersion == previouslyPublishedData.HostDocumentVersion results in a no-op publish.